builds/checkin/dotnet.yaml

```diff
@@ -67,3 +67,44 @@ jobs:
           testRunner: VSTest
           testResultsFiles: '**/TestResults/*.trx'
         condition: succeededOrFailed()
+
+################################################################################
+  - job: code_coverage
+################################################################################
+    displayName: Code coverage
+    variables:
+      coverage.goal: 64
+    pool:
+      vmImage: windows-2019
+    steps:
+      - task: DotNetCoreCLI@2
+        displayName: Build test projects
+        inputs:
+          command: build
+          projects: edge-*\test\Microsoft.Azure.Devices.*Test\Microsoft.Azure.Devices.*Test.csproj
+          arguments: '-o target'
+      - task: VSTest@2
+        displayName: Run unit tests with code coverage
+        inputs:
+          testSelector: testAssemblies
+          testAssemblyVer2: |
+            target\Microsoft.Azure*Test.dll
+          testFiltercriteria: Category=Unit
```

Comment:
Since code coverage only runs unit test, what if a feature is added, but it cannot be easily tested by unit test; but integration test is added.  Therefore the code coverage percentage may go below the goal.  Or we agreed that we should always refactor our code to have some unit tests.  Also do we plan to adding more tests to improve test coverage?

Reply:
A couple thoughts about this:
- Integration tests often require secrets, which is why they don't run in the checkin build. So I can't measure integration tests here.
- I'm only interested in understanding our coverage for unit tests because unit tests are the primary means of "covering" all our features with tests. In most cases, if something can't easily be tested with unit tests it should be refactored until it can. Integration tests are for testing our integration with isolated external systems. I know it's common to write an integration test when unit testing is hard, but I don't want to encourage that behavior.

Reply:
I am on your side.